client/bin/cli.js

```diff
@@ -1,18 +1,49 @@
 #!/usr/bin/env node
 
-import { join, dirname } from "path";
-import { fileURLToPath } from "url";
-import handler from "serve-handler";
+import fs from "fs";
 import http from "http";
+import { dirname, join } from "path";
+import handler from "serve-handler";
+import { fileURLToPath } from "url";
 
 const __dirname = dirname(fileURLToPath(import.meta.url));
 const distPath = join(__dirname, "../dist");
 
 const server = http.createServer((request, response) => {
-  return handler(request, response, {
-    public: distPath,
-    rewrites: [{ source: "/**", destination: "/index.html" }],
-  });
+  // Check if this is a request for index.html (either directly or via SPA routing)
+  if (
+    request.url === "/" ||
+    request.url === "/index.html" ||
+    !request.url.includes(".")
+  ) {
+    const indexPath = join(distPath, "index.html");
+    fs.readFile(indexPath, "utf-8", (err, data) => {
+      if (err) {
+        response.statusCode = 500;
+        response.end(`Error loading index.html: ${err.message}`);
+        return;
+      }
+
+      // Create a runtime config object with environment variables
+      const runtimeConfig = {
+        MCP_PROXY_FULL_ADDRESS: process.env.MCP_PROXY_FULL_ADDRESS || "",
```

Comment:
`MCP_PROXY_FULL_ADDRESS`  and `MCP_PROXY_PORT` should be [ConfigItem](https://github.com/modelcontextprotocol/inspector/blob/main/client/src/lib/configurationTypes.ts#L1)s.